### ISSUE TITLE
Incorrect snapshot in tabs tray after clearing search tabs

### ISSUE THREAD

**Issue by SimonBasca** [mozilla-mobile/firefox-ios#4969]
Build: 17.0 (15191)
Device: iPad Air 2
iOS: 11.4

Steps to reproduce:
1. Open an empty tab
2. Open a New Tab
3. Go to twitter.com
4. Open tabs tray
5. Tap on the "Search Tabs" field
6. Type 'Home' and delete the typed word

Actual results:
- Two twitter.com tabs appear in tabs tray. 

Note:
- After tapping the 'x' in the 'Search Tabs' field the empty tab appears to be displayed correct.

**Comment by KrystynaKruchkovska** [comment]
I am able to reproduce it with step 6 modified a bit, when deleting the typed word I have to do it quite quickly letter by letter.
The problem is the we reload the collectionView of tabs, it's not fully reloaded . The only fix that I managed to find is setting flag `evenIfHidden` to `true` when calling metod `refreshStore`.
It doesn't work perfectly because we have an effect of blinking as described in comment in this method. I'm attaching the video how it works now. I've also checked that it doesn't make tabs screenshot blink in situation when this bug doesn't occur which is also in the video.

[4696.mov.zip](https://github.com/mozilla-mobile/firefox-ios/files/3848330/4696.mov.zip)

I'm opening the pull request with this fix.
